src/node.cc

```diff
@@ -1922,10 +1965,13 @@ void MemoryUsage(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
   size_t rss;
-  int err = uv_resident_set_memory(&rss);
-  if (err) {
-    return env->ThrowUVException(err, "uv_resident_set_memory");
+  int err;
+  {
+    ScopedLock::Mutex lock(&process_mutex);
+    err = uv_resident_set_memory(&rss);
```

Comment:
I _think_ this should be MT-safe on all platforms.  I would suggest dropping the lock and forwarding all bug reports to libuv/libuv. :-)